model/task/task.go

```diff
@@ -2573,20 +2587,37 @@ func ArchiveMany(tasks []Task) error {
 		}
 
 		taskColl := evergreen.GetEnvironment().DB().Collection(Collection)
-		_, err = taskColl.UpdateMany(ctx, bson.M{
-			IdKey: bson.M{
-				"$in": taskIds,
-			},
-		},
-			bson.M{
-				"$unset": bson.M{
-					AbortedKey:   "",
-					AbortInfoKey: "",
+		if execution == -1 {
```

Comment:
I'm not sure we need this if/else statement. In the if case we have inc, in the else statement we have set "execution + 1" which is what $inc means.

Reply:
The logic was largely rewritten to not depend on any magic numbers, booleans, or anything. Using Archive and ArchiveMany should now work as intended for different use cases that we use in the codebase (restart version, etc).